### ISSUE TITLE
DCG push back not implemented, will assert rules for (',')//2

### ISSUE THREAD

**Issue by jburse** [jariazavalverde/tau-prolog#70]
I made a little test with DCG rules that have push backs. It should either
translate them correctly or I guess otherwise it would be better to issue 
an error not yet implemented.

![image](https://user-images.githubusercontent.com/5411487/50870709-7091ed80-13b9-11e9-9ad6-014f806b1280.png)

**Comment by jburse** [comment]
That it asserts something for (',')/2 is seen here, when I go to
the transformation tab, I see (',')/4 rules:

![image](https://user-images.githubusercontent.com/5411487/50870728-88697180-13b9-11e9-8215-084b23712e79.png)

**Comment by jburse** [comment]
This is a pitty, since running these DCG has this result:

![image](https://user-images.githubusercontent.com/5411487/50870803-db432900-13b9-11e9-962d-8f23effe1c5c.png)

**Comment by jburse** [comment]
Here is the test code in textual form:

```
emojis, "\x1f642\" --> ":-)", !, emojis.
emojis, "\x1f626\" --> ":-(", !, emojis.
emojis, "\x1f609\" --> ";-)", !, emojis.
emojis, "\x1f622\" --> ":'(", !, emojis.
emojis, "\x1f61b\" --> ":-P", !, emojis.
emojis, [X] --> [X], !, emojis.
emojis --> [].

atom_emojis(X, Y) :-
   atom_codes(X, L),
   emojis(L, R),
   atom_codes(Y, R).
```

**Comment by jariazavalverde** [comment]
Hi @jburse.

Thanks for your report. Solved in c985aa31e6f67388e5e265cf706b18af00186769!

![pushback](https://user-images.githubusercontent.com/5055295/66707954-4f479e80-ed49-11e9-8701-5929b855cbe7.png)

(Still having issues with UTF-8 encoding).
